python/pyiceberg/catalog/sql.py

```diff
@@ -0,0 +1,441 @@
+        )
+        io = load_file_io(properties=self.properties, location=metadata_location)
+        self._write_metadata(metadata, io, metadata_location)
+
+        with Session(self.engine) as session:
+            try:
+                session.add(
+                    IcebergTables(
+                        catalog_name=self.name,
+                        table_namespace=database_name,
+                        table_name=table_name,
+                        metadata_location=metadata_location,
+                        previous_metadata_location=None,
+                    )
+                )
+                session.commit()
+            except IntegrityError as e:
+                raise TableAlreadyExistsError(f"Table {database_name}.{table_name} already exists") from e
+
+        return self.load_table(identifier=identifier)
+
+    def load_table(self, identifier: Union[str, Identifier]) -> Table:
+        """Loads the table's metadata and returns the table instance.
+
+        You can also use this method to check for table existence using 'try catalog.table() except NoSuchTableError'.
+        Note: This method doesn't scan data stored in the table.
+
+        Args:
+            identifier (str | Identifier): Table identifier.
+
+        Returns:
+            Table: the table instance with its metadata.
+
+        Raises:
+            NoSuchTableError: If a table with the name does not exist.
+        """
+        database_name, table_name = self.identifier_to_database_and_table(identifier, NoSuchTableError)
+        with Session(self.engine) as session:
+            stmt = select(IcebergTables).where(
+                IcebergTables.catalog_name == self.name,
+                IcebergTables.table_namespace == database_name,
+                IcebergTables.table_name == table_name,
+            )
+            result = session.scalar(stmt)
+            if result:
+                return self._convert_orm_to_iceberg(result)
+            raise NoSuchTableError(f"Table does not exist: {database_name}.{table_name}")
+
+    def drop_table(self, identifier: Union[str, Identifier]) -> None:
+        """Drop a table.
+
+        Args:
+            identifier (str | Identifier): Table identifier.
+
+        Raises:
+            NoSuchTableError: If a table with the name does not exist.
+        """
+        database_name, table_name = self.identifier_to_database_and_table(identifier, NoSuchTableError)
+        with Session(self.engine) as session:
+            deleted_rows = (
+                session.query(IcebergTables)
+                .where(
+                    IcebergTables.catalog_name == self.name,
+                    IcebergTables.table_namespace == database_name,
+                    IcebergTables.table_name == table_name,
+                )
+                .delete()
+            )
+            if deleted_rows < 1:
+                raise NoSuchTableError(f"Table does not exist: {database_name}.{table_name}")
+            session.commit()
```

Comment:
If you prefer to have the `delete(IcebergTables)` pattern here as well, the following works for me:
```suggestion
        database_name, table_name = self.identifier_to_database_and_table(identifier, NoSuchTableError)
        with Session(self.engine) as session:
            res = session.execute(
                delete(IcebergTables)
                .where(
                    IcebergTables.catalog_name == self.name,
                    IcebergTables.table_namespace == database_name,
                    IcebergTables.table_name == table_name,
                )
            )
            session.commit()
        if res.rowcount < 1:
            raise NoSuchTableError(f"Table does not exist: {database_name}.{table_name}")
```